### PR TITLE
stomach contents 2 - BMR and activity levels

### PR DESCRIPTION
**Pull request by KorGgenT**
#### Summary

```SUMMARY: Features "Player now burns calories when doing activities"```
#### Purpose of change

This is a continuation of the stomach contents system that moved "hunger" into a kcal storage. I have calculated BMR with @I-am-Erk 's help, and @DracoGriffin has been kind enough to do research on the activity levels of the current activities we have in the game. This PR makes it so that you burn a variable amount calories, based on your activity level. Since there is not currently a way for the player to input age and height for their character, an average is assumed. 
#### Describe the solution

Since weight is not directly kept track of in this system, and weight is one of the factors needed to calculate Basal Metabolic Rate, we instead calculate Body Mass Index from the assumed height and percent_kcal(). 
The activity level of the player is set whenever the player is in an activity, and it resets up to 30 minutes after the end of the activity. This way, we can use the elevated activity level for the next calculations of calorie burn, instead of having a way to cheat the system by doing short bursts of intense activity. However, this does mean that activities have a higher minimum calorie burned than IRL.
#### Additional context

Effects for obesity, overweight, and underweight shall be in a separate PR.
[Here is a spreadsheet with a few value calculations.](https://docs.google.com/spreadsheets/d/1f0LQoXrcBMHoKSfBKH78RFb063qRMLz2-xQ2Q5bvLoM/edit?usp=drive_web&ouid=115426823397702527244)

ALSO: Refer to #29528 for related comments. this PR was reopened due to reaching 30,000

### CODE REVIEW

**Comment by Ilysen** [comment]
I'm not sure if that's put in here, since I can't into code that well, but I heard a good suggestion that when you're filling up after just eating, you gain something next to your hunger listing like there is for body temperature, so it'd display as:

`Hungry (Sating)` or `Hungry (Filling)`.

Could that work as something to add? I think it could improve transparency to the player if something like this isn't already done, since I heard that it can be a little difficult to determine if the food you've eaten has had any effect.

I do like the changes towards making the hunger levels more sensible instead of just increasing them statically, though.

**Comment by Mecares** [comment]
I was thinking maybe stats should increase or decrease the calories you burn. Those huge mountains of muscles need to be maintained after all and less trained characters might need less.

**Comment by Little-119** [comment]
Are the size mutations (going to be) accounted for?

**Comment by JeanLucVanDamme** [comment]
Accounting for size mutations would add some meaningful tradeoffs. It might seem OP to become freakishly strong, but the catch is, now you need way more calories.

All other things being equal, this is true within-species. For humans, bodybuilders must consume far more than the average person to build and maintain muscle.

**Comment by JeanLucVanDamme** [comment]
@KorGgenT, what do you think about allowing certain traits to change a character's assumed (calculated) weight, and thus their baseline BMR at a healthy, normal weight?

You might expect Frail characters to weigh less. They might be frail because of smaller, less dense bone structure. This is definitely a trait that varies between people, likely along a normal distribution, and the differences in some cases can be dramatic. Contrast the frames of Woody Allen vs that guy who plays Jamie Lannister in Game of Thrones.

Here's an interesting study that's consistent with "stronger bones = being tougher":
https://www.ncbi.nlm.nih.gov/pubmed/30356456

In short, I think your recent additions allow some trait choices to become more meaningful and interesting. I bet players would also be more satisfied with the constraints of a BMR system if trait choices gave them more input/control over relevant tradeoffs. So it could be worth exploring possibilities like this.

**Comment by KorGgenT** [comment]
> I was thinking maybe stats should if higher or lower than average increase or decrease the calories you burn. Those huge mountains of muscles need to be maintained after all and less trained characters might need less.
> 
> Different stats could have different impact from high to low:
> Strenght
> Dexterity
> Intelligence
> Perception



> Are the size mutations (going to be) accounted for? Since those obviously change the player character's height. Probably width too (unless characters with the Tiny mutation are pancakes and characters with the Freakishly Huge mutation are slenderman),

I agree with these things. I also think they should be a separate PR, and the numbers should be looked into.

**Comment by kevingranade** [comment]
Trying to merge, maybe don't make activity level mandatory until after the initial merge, as it is I have to manually fix it up if I merge at the same time as anything that adds an activity.

**Comment by kevingranade** [comment]
I'm getting indefinite hangs in the starve test, I had a bunch of other stuff merged but I doubt it was from them.

**Comment by halfahermit** [comment]
Does activity level affect thirst/fatigue/stamina, or only kcal rate?

**Comment by KorGgenT** [comment]
> Does activity level affect thirst/fatigue/stamina, or only kcal rate?

this PR only deals with kcals burned due to activity level.

**Comment by I-am-Erk** [comment]
The BMR formula looks right, and your spreadsheet calculations for BMR at various stored calories / BMIs seems good. 

I'd like to see some test calculations of what ~12-16 hours of light activity plus 8 hours of sleep looks like, as well as maybe 8 hours of medium activity / 8 hours light / 8 hours rest. Those are pretty normal "busy day" activity levels that would help to check the maths and normalize things to make sure it feels about right for real people.

**Comment by Midaychi** [comment]
ACT_HAND_CRANK needs to be updated for this commit, as it as an undefined activity level now.
